doc/test-framework/writing-e2e-tests.md

```diff
@@ -101,19 +101,26 @@ The next step is to create a TestCtx for the current test and defer its cleanup
 
 ```go
 ctx := framework.NewTestCtx(t)
-defer ctx.Cleanup(t)
+defer ctx.Cleanup()
 ```
 
-Now that there is a TestCtx, the test's kubernetes resources (specifically the test namespace,
-RBAC, and Operator deployment) can be initialized:
+Now that there is a `TestCtx`, the test's kubernetes resources (specifically the test namespace,
+Service Account, RBAC, and Operator deployment in `local` testing; just the Operator deployment
+in `cluster` testing) can be initialized:
 
 ```go
-err := ctx.InitializeClusterResources()
+err := ctx.InitializeClusterResources(&framework.CleanupOptions{Timeout: time.Duration * 5, RetryInterval: time.Duration * 1})
 if err != nil {
     t.Fatalf("failed to initialize cluster resources: %v", err)
 }
 ```
 
+The `InitializeClusterResources` function uses the `CreateWithCleanup` function to creates the resources provided
+in your namespaced manifest. `CreateWithCleanup` use the controller-runtime's client to create resources and then
+creates a cleanup function that is called by `ctx.Cleanup` which deletes the resource and then waits for the resource to be
+fully deleted before returning. This is configurable with `CleanupOptions`. If the `CleanupOptions` argument is set
+to `nil`, the cleanup function simply calls the delete function and returns without waiting.
```

Comment:
I know this is a bit subjective but how about rewording the above to:

`CreateWithCleanup` will create a resource and adds a cleanup function for that resource to the current context.
All cleanup functions for the context will run when `ctx.Cleanup()` called.
The cleanup function will delete the resource and then optionally poll and wait until the resource is completely removed from the apiserver. 
The `CleanupOptions` argument configures the polling timeout and retry period for the cleanup. 
If the `CleanupOptions` argument is set to `nil` the cleanup function will only delete and not wait for the resource to be removed from the apiserver.